utils/include/WindowsUtils.cpp

```diff
@@ -114,18 +115,43 @@ namespace PlatformUtils
 		return success;
 	}
 
-	bool TryGetLogicalDisksNames(char* array_to_write, 
-		const unsigned short c_size_of_buffer_for_api)
+	bool TryGetLogicalDisksNames(std::vector<std::string>& all_disks_names)
 		{
+
+		const unsigned short c_size_of_buffer_for_api = 1024;
+		//We just skip some chars
+		const unsigned short number_of_chars_need_miss = 1U;
+		char container_all_disks_names[c_size_of_buffer_for_api +
+			c_size_of_buffer_for_api] = {};
+		
 		DWORD buffer_size = c_size_of_buffer_for_api;
 		DWORD is_created_correct = GetLogicalDriveStrings(buffer_size,
-			LPSTR(array_to_write));
+			LPSTR(container_all_disks_names));
+
 		if (is_created_correct > 0 &&
 			is_created_correct <= c_size_of_buffer_for_api)
+		{
+			char* variable_for_checking_names = container_all_disks_names;
```

Comment:
Where do you free up memory?

Reply:
I don't use dynamic memory here. API just fill array of chars